src/main/java/racing/domain/RacingGame.java

```diff
@@ -1,81 +1,46 @@
 package racing.domain;
 
 import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
+import java.util.stream.Collectors;
+import racing.domain.car.Car;
+import racing.domain.car.Cars;
+import racing.domain.car.Position;
+import racing.domain.startegy.MovingStrategy;
+import racing.domain.startegy.RandomMovingStrategy;
 
 public class RacingGame {
 
-    private static final int RANGE_OF_RANDOM_NUMBER = 10;
+    private static final String COMMA = ", ";
 
-    private final List<Car> cars;
+    private final MovingStrategy movingStrategy;
+    private final List<List<Car>> raceStates = new ArrayList<>();
+    private final Cars cars;
 
-    public RacingGame() {
-        this.cars = new ArrayList<>();
+    public RacingGame(Cars cars) {
+        movingStrategy = new RandomMovingStrategy();
+        this.cars = cars;
     }
 
-    public void startGame() {
-        try {
-            final List<String> nameList = UserInput.getCarNames();
-            for (String name : nameList) {
-                cars.add(new Car(name));
-            }
-            checkCarCount(cars);
-
-            final int tryNumber = UserInput.getTryNumber();
-            startRacing(tryNumber);
-            showGameResult();
-        } catch (IllegalArgumentException e) {
-            System.out.println(e.getMessage());
-            startGame();
-        }
-    }
-
-    private static void checkCarCount(List<Car> cars) {
-        if (cars.isEmpty()) {
-            throw new IllegalArgumentException("[ERROR] 최소 1대 이상의 자동차가 필요합니다.");
-        }
-    }
-
-    private void startRacing(final int tryNumber) {
-        System.out.println("\n실행 결과");
-        for (int i = 0; i < tryNumber; i++) {
-            driveCars();
-            System.out.println();
-        }
+    public static RacingGame registerCars(List<String> names) {
+        return new RacingGame(new Cars(names, 0));
     }
 
-    private void driveCars() {
-        for (Car car : cars) {
-            car.drive(generateRandomNumber());
+    public void race(final TryNumber tryNumber) {
+        while (tryNumber.hasNextStep()) {
+            raceStates.add(cars.driveCars(movingStrategy));
+            tryNumber.nextStep();
         }
     }
```

Comment:
`raceStates`라는 상태 값을 둘 필요 없이 `race` 메서드 내에 변수로 두고 이 변수를 반환하면 어떨까요? 🤔 

```suggestion
    public List<Cars> race(final TryNumber tryNumber) {
        List<Cars> raceStates = new ArrayList<>();
        while (tryNumber.hasNextStep()) {
            raceStates.add(cars.driveCars(movingStrategy));
            tryNumber.nextStep();
        }
        return raceStates;
    }
```

Reply:
와... 리뷰어님 말씀대로 굳이 인스턴스 변수를 두지 않고, 하나의 메서드로 해결할 수 있네요! 오늘도 많이 배워갑니다 😁